src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/Perf.Base64.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Base64
+    {
+        private ArrayBufferWriter<byte> _arrayBufferWriter;
+        private byte[] _data;
+
+        [Params(10, 100, 1000)]
+        public int NumberOfBytes { get; set; }
+
+        [Params(true, false)]
+        public bool Formatted;
+
+        [Params(true, false)]
+        public bool SkipValidation;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _arrayBufferWriter = new ArrayBufferWriter<byte>();
+            _data = ValuesGenerator.Array<byte>(NumberOfBytes);
+        }
+
+        [Benchmark]
+        public void WriteByteArrayAsBase64()
+        {
+            _arrayBufferWriter.Clear();
+            using (var json = new Utf8JsonWriter(_arrayBufferWriter, new JsonWriterOptions { Indented = Formatted, SkipValidation = SkipValidation }))
+            {
+                json.WriteBase64String("data", _data);
```

Comment:
As this code is written now, this will throw at runtime:
```
System.InvalidOperationException
  HResult=0x80131509
  Message=Cannot write a JSON property within an array or as the first JSON token. Current token type is 'None'.
  Source=System.Text.Json
  StackTrace:
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource, Int32 currentDepth, Byte token, JsonTokenType tokenType)
   at System.Text.Json.Utf8JsonWriter.WriteBase64ByOptions(ReadOnlySpan`1 propertyName, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonWriter.WriteBase64String(ReadOnlySpan`1 propertyName, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonWriter.WriteBase64String(String propertyName, ReadOnlySpan`1 bytes)
...
```

I'll correct this.

Didn't realize that yesterday, as the API wasn't in the SDK :wink:.

Reply:
Fixed with e771ace5986fd2002d4497c8e4594f52001c47b1